src/utils/packet.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const IlpPacket = require('ilp-packet')
+const assert = require('assert')
+const debug = require('debug')('ilp:packet')
+
+const serialize = IlpPacket.serializeIlpPayment
+const parse = (packet) => {
+  try {
+    return IlpPacket.deserializeIlpPayment(packet)
+  } catch (e) {
+    debug('error while parsing packet: ' + e.message)
+    return undefined
+  }
+}
+
+function getFromTransfer (transfer) {
+  assert(transfer, 'transfer must be defined. got: ' + transfer)
+  assert(typeof transfer === 'object', 'got invalid transfer: ' + transfer)
+  assert(Buffer.isBuffer(transfer.ilp),
```

Comment:
According to https://github.com/interledger/rfcs/pull/165 `transfer.ilp` will be a string